### DISCUSSION TITLE
Edit/modify a step file - possible?

### DISCUSSION THREAD

**Discussion by sonicdee** [CadQuery/cadquery#1358] [Q&A]
Hello, 
it's easy to import a Step file, currently using cq-editor on win but anyway...
import cadquery as cq
result = cq.importers.importStep('P:\ath\to\File.stp')

But can we also edit the file, so go through the single items, like plane/circle/points, 
and modify them, and export afterwards ?

**Comment by jmwright** [comment]
You can modify step files after importing them.

```python
import cadquery as cq

filepath = "/home/jwright/Downloads/ex.step"

ex = cq.Workplane().box(10, 10, 10)

cq.exporters.export(ex, filepath)

ex2 = cq.importers.importStep(filepath)

ex2 = ex2.faces(">Z").workplane(invert=True).circle(3.0).extrude(5.0, combine='cut')
ex2 = ex2.edges("<Z and |Y").fillet(1.0)

show_object(ex2)
```

**Reply by jmwright** [reply]
I should also add that when you round-trip STEP files created by non-CQ (or OpenCASCADE) software, they may be exported differently than the original tool exported them. The file size may change, and if you open the text file in an editor you may see differences.
